dotnet/src/Experimental/Process.LocalRuntime/LocalStep.cs

```diff
@@ -76,6 +75,11 @@ public LocalStep(KernelProcessStepInfo stepInfo, Kernel kernel, string? parentPr
     /// </summary>
     internal string Id => this._stepInfo.State.Id!;
 
+    /// <summary>
+    /// An event filter that can be used to intercept events emitted by the step.
+    /// </summary>
+    internal ProcessEventProxy? EventProxy { get; init; }
```

Comment:
Following up on my early comment, it looks odd that the description uses filter but the name uses proxy.

If I understand the usage, this is a delegate that gets invoked before emitting an event, which if false is returned will prevent the event being emitted.

Consider rethinking the naming here, maybe something like:

```csharp
ShouldEmitEventHandler? ShouldEmitEvent { ... }
ShouldEmitEventDelegate? ShouldEmitEvent { ... }

Reply:
Updated comment.  Creating issue to continue evaluation of the internal event-proxy feature.  We're considering how this might apply outside of map-step, which will provide additional data-points to drive evolution.